torch/utils/dlpack.py

```diff
@@ -22,33 +22,78 @@ class DLDeviceType(enum.IntEnum):
 
 torch._C._add_docstr(to_dlpack, r"""to_dlpack(tensor) -> PyCapsule
 
-Returns a DLPack representing the tensor.
+Returns an opaque object (a "DLPack capsule") representing the tensor.
+
+.. note::
+  ``to_dlpack`` is a legacy DLPack interface. The capsule it returns
+  cannot be used for anything in Python other than use it as input to
+  ``from_dlpack``. The more idiomatic use of DLPack is to call
+  ``from_dlpack`` directly on the tensor object - this works when that
+  object has a ``__dlpack__`` method, which PyTorch and most other
+  libraries indeed have now.
+
+.. warning::
+  Only call ``from_dlpack`` once per capsule produced with ``to_dlpack``.
+  Behavior when a capsule is consumed multiple times is undefined.
```

Comment:
Nitpick: technically it's well defined (an error would be raised if the capsule has been consumed with its name changed) long before we settled on `__dlpack__`, but I'm fine leaving it as is.